src/main/scala/chisel3/util/Counter.scala

```diff
@@ -19,27 +19,50 @@ import chisel3.internal.naming.chiselName  // can't use chisel3_ version because
   *     ...
   *   }
   * }}}
-  *
-  * @param n number of counts before the counter resets (or one more than the
-  * maximum output value of the counter), need not be a power of two
```

Comment:
It would be nice to have ScalaDoc with the new range API

Reply:
Absolutely, I want to do a follow-up PR to update all the ScalaDoc for the Counter API.

I originally had those changes in here, but removed them to keep this PR more focused.